pkg/config/patch_test.go

```diff
@@ -0,0 +1,148 @@
+                    }
+                  }
+                }`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"added": "via-patch"}),
+		},
+		{
+			name:   "add labels to deployment (JSON6902, yaml)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test"),
+			patches: []v1alpha1.Patch{{
+				Kind: "Deployment",
+				Patch: json.RawMessage(`
+  op: add
+  path: /metadata/labels
+  value: 
+    added: via-patch`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"added": "via-patch"}),
+		},
+		{
+			name:   "add labels to deployment (JSON6902, json)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test"),
+			patches: []v1alpha1.Patch{{
+				Kind:  "Deployment",
+				Patch: json.RawMessage(`{"op": "add", "path": "/metadata/labels", "value":{"added":"via-patch"}}`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"added": "via-patch"}),
+		},
+		{
+			name:   "add labels to deployment without affecting existing labels (JSON6902, json)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"initial": "label"}),
+			patches: []v1alpha1.Patch{{
+				Kind:  "Deployment",
+				Patch: json.RawMessage(`{"op": "add", "path": "/metadata/labels", "value":{"added":"via-patch"}}`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(
+				map[string]string{
+					"initial": "label",
+					"added":   "via-patch",
+				},
+			),
+		},
+		{
+			name:   "add labels to deployment without affecting existing labels (smp, json)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"initial": "label"}),
+			patches: []v1alpha1.Patch{{
+				Kind: "Deployment",
+				Patch: json.RawMessage(`{
+                  "metadata": {
+                    "labels": {
+                      "added": "via-patch"
+                    }
+                  }
+                }`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(
+				map[string]string{
+					"initial": "label",
+					"added":   "via-patch",
+				},
+			),
+		},
+		{
+			name:   "add labels to deployment with a wildcard match (smp, yaml)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test"),
+			patches: []v1alpha1.Patch{{
+				Kind: "*",
+				Patch: json.RawMessage(`
+metadata:
+  labels:
+    added: via-patch`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"added": "via-patch"}),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if _, _, err := ApplyPatches(tt.object, tt.patches); (err != nil) != tt.wantErr {
+				t.Errorf("ApplyPatches() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			require.Equal(t, tt.want, tt.object)
+		})
+	}
+}
+
+// TODO: test that all exported methods on `Config` apply patches
```

Comment:
just in case you intended to address this in the PR ☝🏻

Reply:
Yep, it's in `config_test.go`